File: c++/leetcode/806. Number of Lines To Write String.cpp
class Solution {
public:
    vector<int> numberOfLines(vector<int>& widths, string S) {
        int ans = 1, line = 0;
        
        for(auto i : S) {
            auto w = widths[i - 'a'];
            ans = line + w > 100 ? ans + 1 : ans;
            line = line + w > 100 ? w : line + w;
        }
        
        return {ans, line};
    }
};
 

File: javascript/leetcode/806. Number of Lines To Write String.js
/**
 * @param {number[]} widths
 * @param {string} S
 * @return {number[]}
 */
var numberOfLines = function(widths, S) {
    let ans = 1, line = 0
    
    const base = 'a'.charCodeAt(0);
    
    for(let i = 0; i < S.length; ++i) {
        const w = widths[S.charCodeAt(i) - base];
        ans = line + w > 100 ? ans + 1 : ans
        line = line + w > 100 ? w : line + w
    }
    
    return [ans, line]
};
 
